### ISSUE TITLE
Expose floating arithmetics in RzNum/`rz-ax` tool

### ISSUE THREAD

**Issue by XVilka** [rizinorg/rizin#3035]
Blocked by https://github.com/rizinorg/rizin/issues/3455

Currently, we have RzFloat bitvector API that will be used in RzIL. It would be nice to have it exposed also in the Rizin's calculator - `rz-ax`, and some of the RzNum APIs.

- `librz/util/float/*`
- `librz/include/rz_util/rz_float.h`

**Comment by Heersin** [comment]
Expose both bitv and float API, and assign new options for them would be nice. For example
- `-z` : convert number as bitvector representation, `-Zu/-Zi` bitvector -> unsigned/signed integer, `-Zf` bitvector as signed -> float, `-Zp` bitvector as unsigned -> float
- `-zf`: caculate float using bitvector arithmetic, and `-zi` caculate number using bitvector arithmetic

**Comment by XVilka** [comment]
Also `0b` literals should be treated as bitvectors by default, to solve this problem: https://github.com/rizinorg/rizin/issues/3331
